IL2CPP support?

Does this package support being compiled with IL2CPP?  I just tried doing that but got a bunch of errors like this:

```
Assets\Plugins\HoloToolkit\SpatialMapping\Scripts\RemoteMapping\MeshSaver.cs(10,24): error CS0234: The type or namespace name 'Tasks' does not exist in the namespace 'System.Threading' (are you missing an assembly reference?)
Assets\Plugins\HoloToolkit\SpatialMapping\Scripts\SpatialProcessing\SurfaceMeshesToPlanes.cs(11,24): error CS0234: The type or namespace name 'Tasks' does not exist in the namespace 'System.Threading' (are you missing an assembly reference?)
Assets\Plugins\HoloToolkit\CrossPlatform\Scripts\Reflection\ReflectionExtensions.cs(90,59): error CS0246: The type or namespace name 'TypeInfo' could not be found (are you missing a using directive or an assembly reference?)
```

Hi, sorry to reopen the issue, but at my side it still not working. I have just created a new project and added the Holotoolkit as explained in the GetStarted.md (Using the ClickThroughExample). But when I try to compile i have 3 errors


```
Assets\HoloToolkit\SpatialMapping\Scripts\RemoteMapping\MeshSaver.cs(162,53): error CS1061: 'StorageFile' does not contain a definition for 'OpenStreamForReadAsync' and no extension method 'OpenStreamForReadAsync' accepting a first argument of type 'StorageFile' could be found (are you missing a using directive or an assembly reference?)
Assets\HoloToolkit\SpatialMapping\Scripts\RemoteMapping\MeshSaver.cs(189,53): error CS1061: 'StorageFile' does not contain a definition for 'OpenStreamForWriteAsync' and no extension method 'OpenStreamForWriteAsync' accepting a first argument of type 'StorageFile' could be found (are you missing a using directive or an assembly reference?)
Error building Player because scripts had compiler errors`
```
I have selected .Net 4.6 for the API compatibility level and used Holotoolkit -> build Windows for building

Sorry for my bad english, it's not my main language.

--- a/Assets/HoloToolkit/SpatialMapping/Prefabs/RemoteMapping.prefab
+++ b/Assets/HoloToolkit/SpatialMapping/Prefabs/RemoteMapping.prefab
@@ -16,13 +16,13 @@ GameObject:
   m_ObjectHideFlags: 0
   m_PrefabParentObject: {fileID: 0}
   m_PrefabInternal: {fileID: 100100000}
-  serializedVersion: 4
+  serializedVersion: 5
   m_Component:
-  - 4: {fileID: 4000013382681936}
-  - 114: {fileID: 114000013950815180}
-  - 114: {fileID: 114000010791004466}
-  - 114: {fileID: 114000012256053342}
-  - 114: {fileID: 114000013862356290}
+  - component: {fileID: 4000013382681936}
+  - component: {fileID: 114000013950815180}
+  - component: {fileID: 114000010791004466}
+  - component: {fileID: 114000012256053342}
+  - component: {fileID: 114000013862356290}
   m_Layer: 0
   m_Name: RemoteMapping
   m_TagString: Untagged
@@ -67,6 +67,8 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: e0e40e7a40c41984bbec8aaa64292522, type: 3}
   m_Name: 
   m_EditorClassIdentifier: 
+  RemoteMappingKey: 110
+  SendMeshesKeyword: send meshes
 --- !u!114 &114000013862356290
 MonoBehaviour:
   m_ObjectHideFlags: 1
@@ -79,6 +81,8 @@ MonoBehaviour:
   m_Name: 
   m_EditorClassIdentifier: 
   MeshFileName: roombackup
+  LoadFileKey: 108
+  SaveFileKey: 107
 --- !u!114 &114000013950815180
 MonoBehaviour:
   m_ObjectHideFlags: 1

--- a/Assets/HoloToolkit/SpatialMapping/Scripts/RemoteMapping/MeshSaver.cs
+++ b/Assets/HoloToolkit/SpatialMapping/Scripts/RemoteMapping/MeshSaver.cs
@@ -9,6 +9,7 @@ using UnityEngine;
 #if !UNITY_EDITOR && UNITY_METRO
 using System.Threading.Tasks;
 using Windows.Storage;
+using Windows.Storage.Streams;
 #endif
 
 namespace HoloToolkit.Unity.SpatialMapping
@@ -153,13 +154,14 @@ namespace HoloToolkit.Unity.SpatialMapping
         {
             Stream stream = null;
 
-#if !UNITY_EDITOR && UNITY_METRO
+#if !UNITY_EDITOR && UNITY_WSA
             Task<Task> task = Task<Task>.Factory.StartNew(
                             async () =>
                             {
                                 StorageFolder folder = await StorageFolder.GetFolderFromPathAsync(folderName);
                                 StorageFile file = await folder.GetFileAsync(fileName);
-                                stream = await file.OpenStreamForReadAsync();
+                                IRandomAccessStreamWithContentType randomAccessStream = await file.OpenReadAsync();
+                                stream = randomAccessStream.AsStreamForRead();
                             });
             task.Wait();
             task.Result.Wait();
@@ -180,13 +182,14 @@ namespace HoloToolkit.Unity.SpatialMapping
         {
             Stream stream = null;
 
-#if !UNITY_EDITOR && UNITY_METRO
+#if !UNITY_EDITOR && UNITY_WSA
             Task<Task> task = Task<Task>.Factory.StartNew(
                             async () =>
                             {
                                 StorageFolder folder = await StorageFolder.GetFolderFromPathAsync(folderName);
                                 StorageFile file = await folder.CreateFileAsync(fileName, CreationCollisionOption.ReplaceExisting);
-                                stream = await file.OpenStreamForWriteAsync();
+                                IRandomAccessStream randomAccessStream = await file.OpenAsync(FileAccessMode.ReadWrite);
+                                stream = randomAccessStream.AsStreamForWrite();
                             });
             task.Wait();
             task.Result.Wait();

--- a/Assets/HoloToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs
+++ b/Assets/HoloToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs
@@ -89,6 +89,12 @@ namespace HoloToolkit.Unity.SpatialMapping
                     // Pass the data to the mesh serializer. 
                     List<Mesh> meshes = new List<Mesh>(SimpleMeshSerializer.Deserialize(dataBuffer));
 
+                    if (meshes.Count > 0)
+                    {
+                        // Use the network-based mapping source to receive meshes in the Unity editor.
+                        SpatialMappingManager.Instance.SetSpatialMappingSource(this);
+                    }
+
                     // For each mesh, create a GameObject to render it.
                     for (int index = 0; index < meshes.Count; index++)
                     {

--- a/Assets/HoloToolkit/SpatialMapping/Tests/Scenes/RemoteMapping.unity
+++ b/Assets/HoloToolkit/SpatialMapping/Tests/Scenes/RemoteMapping.unity
@@ -1,19 +1,19 @@
 %YAML 1.1
 %TAG !u! tag:unity3d.com,2011:
 --- !u!29 &1
-SceneSettings:
+OcclusionCullingSettings:
   m_ObjectHideFlags: 0
-  m_PVSData: 
-  m_PVSObjectsArray: []
-  m_PVSPortalsArray: []
+  serializedVersion: 2
   m_OcclusionBakeSettings:
     smallestOccluder: 5
     smallestHole: 0.25
     backfaceThreshold: 100
+  m_SceneGUID: 00000000000000000000000000000000
+  m_OcclusionCullingData: {fileID: 0}
 --- !u!104 &2
 RenderSettings:
   m_ObjectHideFlags: 0
-  serializedVersion: 7
+  serializedVersion: 8
   m_Fog: 0
   m_FogColor: {r: 0.5, g: 0.5, b: 0.5, a: 1}
   m_FogMode: 3
@@ -25,6 +25,7 @@ RenderSettings:
   m_AmbientGroundColor: {r: 0.047, g: 0.043, b: 0.035, a: 1}
   m_AmbientIntensity: 1
   m_AmbientMode: 0
+  m_SubtractiveShadowColor: {r: 0.42, g: 0.478, b: 0.627, a: 1}
   m_SkyboxMaterial: {fileID: 10304, guid: 0000000000000000f000000000000000, type: 0}
   m_HaloStrength: 0.5
   m_FlareStrength: 1
@@ -37,11 +38,11 @@ RenderSettings:
   m_ReflectionIntensity: 1
   m_CustomReflection: {fileID: 0}
   m_Sun: {fileID: 0}
-  m_IndirectSpecularColor: {r: 0.44692492, g: 0.4967869, b: 0.57508546, a: 1}
+  m_IndirectSpecularColor: {r: 0.44657898, g: 0.4964133, b: 0.5748178, a: 1}
 --- !u!157 &3
 LightmapSettings:
   m_ObjectHideFlags: 0
-  serializedVersion: 7
+  serializedVersion: 11
   m_GIWorkflowMode: 0
   m_GISettings:
     serializedVersion: 2
@@ -53,7 +54,7 @@ LightmapSettings:
     m_EnableBakedLightmaps: 1
     m_EnableRealtimeLightmaps: 1
   m_LightmapEditorSettings:
-    serializedVersion: 4
+    serializedVersion: 9
     m_Resolution: 2
     m_BakeResolution: 40
     m_TextureWidth: 1024
@@ -66,29 +67,50 @@ LightmapSettings:
     m_LightmapParameters: {fileID: 0}
     m_LightmapsBakeMode: 1
     m_TextureCompression: 1
-    m_DirectLightInLightProbes: 1
     m_FinalGather: 0
     m_FinalGatherFiltering: 1
     m_FinalGatherRayCount: 256
     m_ReflectionCompression: 2
+    m_MixedBakeMode: 1
+    m_BakeBackend: 0
+    m_PVRSampling: 1
+    m_PVRDirectSampleCount: 32
+    m_PVRSampleCount: 500
+    m_PVRBounces: 2
+    m_PVRFilterTypeDirect: 0
+    m_PVRFilterTypeIndirect: 0
+    m_PVRFilterTypeAO: 0
+    m_PVRFilteringMode: 0
+    m_PVRCulling: 1
+    m_PVRFilteringGaussRadiusDirect: 1
+    m_PVRFilteringGaussRadiusIndirect: 5
+    m_PVRFilteringGaussRadiusAO: 2
+    m_PVRFilteringAtrousPositionSigmaDirect: 0.5
+    m_PVRFilteringAtrousPositionSigmaIndirect: 2
+    m_PVRFilteringAtrousPositionSigmaAO: 1
   m_LightingDataAsset: {fileID: 0}
-  m_RuntimeCPUUsage: 25
+  m_UseShadowmask: 0
 --- !u!196 &4
 NavMeshSettings:
   serializedVersion: 2
   m_ObjectHideFlags: 0
   m_BuildSettings:
     serializedVersion: 2
+    agentTypeID: 0
     agentRadius: 0.5
     agentHeight: 2
     agentSlope: 45
     agentClimb: 0.4
     ledgeDropHeight: 0
     maxJumpAcrossDistance: 0
-    accuratePlacement: 0
     minRegionArea: 2
-    cellSize: 0.16666667
     manualCellSize: 0
+    cellSize: 0.16666667
+    manualTileSize: 0
+    tileSize: 256
+    accuratePlacement: 0
+    debug:
+      m_Flags: 0
   m_NavMeshData: {fileID: 0}
 --- !u!1001 &920116912
 Prefab:
@@ -221,10 +243,10 @@ GameObject:
   m_ObjectHideFlags: 0
   m_PrefabParentObject: {fileID: 0}
   m_PrefabInternal: {fileID: 0}
-  serializedVersion: 4
+  serializedVersion: 5
   m_Component:
-  - 4: {fileID: 1615710845}
-  - 108: {fileID: 1615710844}
+  - component: {fileID: 1615710845}
+  - component: {fileID: 1615710844}
   m_Layer: 0
   m_Name: Directional Light
   m_TagString: Untagged
@@ -239,7 +261,7 @@ Light:
   m_PrefabInternal: {fileID: 0}
   m_GameObject: {fileID: 1615710843}
   m_Enabled: 1
-  serializedVersion: 7
+  serializedVersion: 8
   m_Type: 1
   m_Color: {r: 1, g: 0.95686275, b: 0.8392157, a: 1}
   m_Intensity: 1
@@ -264,6 +286,8 @@ Light:
   m_Lightmapping: 4
   m_AreaSize: {x: 1, y: 1}
   m_BounceIntensity: 1
+  m_ColorTemperature: 6570
+  m_UseColorTemperature: 0
   m_ShadowRadius: 0
   m_ShadowAngle: 0
 --- !u!4 &1615710845
@@ -275,7 +299,7 @@ Transform:
   m_LocalRotation: {x: 0.40821788, y: -0.23456968, z: 0.10938163, w: 0.8754261}
   m_LocalPosition: {x: 0, y: 3, z: 0}
   m_LocalScale: {x: 1, y: 1, z: 1}
-  m_LocalEulerAnglesHint: {x: 50, y: -30, z: 0}
   m_Children: []
   m_Father: {fileID: 0}
   m_RootOrder: 0
+  m_LocalEulerAnglesHint: {x: 50, y: -30, z: 0}

